### ISSUE TITLE
Make the `at` parameter of `splitobs` accept vector

### ISSUE THREAD

**Issue by innerlee** [JuliaML/MLDataPattern.jl#43]
Currently `splitobs(data, at[, obsdim]) → NTuple` is slow when splitting some data into many many parts.
For example, `@time splitobs(rand(100000), at=ntuple(i->1/10001, 10000))` will take forever to run.

Proposal: make `at` accept vector, and return vector of data.

**Comment by innerlee** [comment]
Anyone?

**Comment by johnnychen94** [comment]
I don't have a use case other than splitting data into more than three parts: train/valid/test.

P.S. If you want to make 10000 batches, you could use `RandomBatches(data, batch_sz, 10000)` instead.

**Comment by innerlee** [comment]
I do not need random batches. I need list of subsets whose disjoint union is the whole dataset.

splitobs has this semantics, but RandomBatches does not.

**Comment by johnnychen94** [comment]
In that case, I think [`eachbatch`](https://mldatapatternjl.readthedocs.io/en/latest/documentation/dataiterator.html#eachbatch)/[`batchview`](https://mldatapatternjl.readthedocs.io/en/latest/documentation/dataview.html#as-vector-of-batches) already works nicely.

**Comment by innerlee** [comment]
Actually not exactly. It drops elements.

```julia
julia> a = collect(1:10)
10-element Array{Int64,1}:
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10

julia> b = batchview(a, 3)
3-element batchview(::Array{Int64,1}, 3, 3, ObsDim.Last()) with eltype SubArray:
 [1, 2, 3]
 [4, 5, 6]
 [7, 8, 9]
```

**Comment by johnnychen94** [comment]
> It drops elements.

Well, yes. However, as long as the data get shuffled and the model gets trained in many epochs, since every sample would be seen, that won't give any visible change the final results.

cref: https://github.com/JuliaML/MLDataPattern.jl/issues/14

**Comment by johnnychen94** [comment]
BTW, I'm skeptical that tuple/vector difference would affect the splitting performance.

**Comment by innerlee** [comment]
How about inferring? One cannot omit samples.

**Comment by innerlee** [comment]
Yeah one cannot easily generate long tuples. So if we change the interface to Vector, things will be easier

**Comment by johnnychen94** [comment]
A second thought. The propose makes sense since the tuple/vector difference is usually not a performance bottleneck for ml tasks.

The long tuple issue is possibly related to https://github.com/JuliaLang/julia/issues/35547 The correct workaround for your example code is: `tuple(fill(1/10001, 10000))`
